libs/api/domains/notifications/src/lib/notifications.model.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Field,
+  GraphQLISODateTime,
+  ID,
+  InputType,
+  ObjectType,
+  registerEnumType,
+} from '@nestjs/graphql'
+import { PageInfoDto } from '@island.is/nest/pagination'
+
+// TODO move to/import from client
+export enum NotificationStatus {
+  READ = 'READ',
+  UNREAD = 'UNREAD',
+}
+
+registerEnumType(NotificationStatus, { name: 'NotificationStatus' })
+
+@ObjectType('NotificationMetadata')
+export class NotificationMetadata {
+  @Field(() => GraphQLISODateTime)
+  sent!: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  read?: Date
+
+  @Field(() => NotificationStatus)
+  status!: NotificationStatus
+}
+
+@ObjectType('NotificationSender')
+export class NotificationSender {
+  @Field()
+  name!: string
+
+  @Field({ nullable: true })
+  logo?: string
+}
+
+@ObjectType('NoticiationRecipient')
+export class NotificationRecipient {
+  @Field()
+  nationalId!: string
+}
+
+@ObjectType('NotificationLink')
+export class NotificationLink {
+  @Field()
+  uri!: string
+}
+
+@ObjectType('NotificationMessage')
+export class NotificationMessage {
+  @Field()
+  title!: string
+
+  @Field()
+  body!: string
+
+  @Field(() => NotificationLink)
+  link!: NotificationLink
+}
+
+@ObjectType('Notification')
+export class Notification {
+  @Field(() => ID)
+  notificationId!: string
+
+  @Field(() => NotificationMetadata)
+  metadata!: NotificationMetadata
+
+  @Field(() => NotificationSender)
+  sender!: NotificationSender
+
+  @Field(() => NotificationRecipient)
+  recipient!: NotificationRecipient
+
+  @Field(() => NotificationMessage)
+  message!: NotificationMessage
+}
+
+@ObjectType('MessageCounts')
```

Comment:
Must prefix all GraphQL object types with the domain name unless the object type contains either singular or plural version of the domain name, see [Code docs](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions)
```suggestion
@ObjectType('NotificationsMessageCounts')
```